xhr/client-faithful/src/main/scala/endpoints4s/xhr/faithful/Endpoints.scala

```diff
@@ -3,14 +3,18 @@ package endpoints4s.xhr.faithful
 import endpoints4s.xhr
 import faithful.{Future, Promise}
 
+import scala.scalajs.js
+
 /** Implements [[xhr.Endpoints]] by using faithful.
   *
   * @group interpreters
   */
 trait Endpoints extends xhr.Endpoints {
 
   /** Maps `Result` to `Future` */
-  type Result[A] = Future[A]
+  abstract class Result[A](val future: Future[A]) {
+    val abort: js.Function0[Unit]
```

Comment:
May I request to change this one to:

```suggestion
    def abort(): Unit
```

For the sake of consistency with the other interpreters?